.github/workflows/tests.yaml

```diff
@@ -0,0 +1,133 @@
+name: Tests
+
+on:
+  push:
+    branches: [ master, 'maint-*' ]
+    paths:
+      - '.github/workflows/tests.yaml'
+      - 'requirements*.txt'
+      - 'setup.py'
+      - 'setup.cfg'
+      - 'MANIFEST.in'
+      - 'pyproject.toml'
+      - 'scripts/**'
+      - 'fiona/**'
+      - 'tests/**'
+  pull_request:
+    branches: [ master, 'maint-*' ]
+    paths:
+      - '.github/workflows/tests.yaml'
+      - 'requirements*.txt'
+      - 'setup.py'
+      - 'setup.cfg'
+      - 'MANIFEST.in'
+      - 'pyproject.toml'
+      - 'scripts/**'
+      - 'fiona/**'
+      - 'tests/**'
+  schedule:
+    - cron:  '0 0 * * 0'
+
+jobs:
+  docker_tests:
+    runs-on: ubuntu-latest
+    name: Docker | GDAL=${{ matrix.gdal-version }} | python=${{ matrix.python-version }}
+    container: osgeo/gdal:ubuntu-small-${{ matrix.gdal-version }}
+    env:
+        DEBIAN_FRONTEND: noninteractive
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.9', '3.10']
```

Comment:
Personally, I'm a fan of testing all (or at least the oldest version of Python that is supported by the project as well as the latest). That would be 3.7 to 3.11.  Same it would be good to include a spread between the oldest supported gdal version and the newest.